e2e/e2e_canary_upgrade_test.go

```diff
@@ -0,0 +1,279 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/stackrox/acs-fleet-manager/fleetshard/pkg/central/operator"
+	"github.com/stackrox/acs-fleet-manager/internal/dinosaur/constants"
+	"github.com/stackrox/acs-fleet-manager/internal/dinosaur/pkg/api/public"
+	"github.com/stackrox/acs-fleet-manager/internal/dinosaur/pkg/services"
+	"github.com/stackrox/acs-fleet-manager/pkg/client/fleetmanager"
+	"github.com/stackrox/rox/operator/apis/platform/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+	ctrlClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	namespace = "acsms"
+)
+
+var _ = Describe("Fleetshard-sync Canary Upgrade", func() {
+
+	BeforeEach(func() {
+		if !runCanaryUpgradeTests {
+			Skip("Skipping canary upgrade test")
+		}
+	})
+
+	if fmEndpointEnv := os.Getenv("FLEET_MANAGER_ENDPOINT"); fmEndpointEnv != "" {
+		fleetManagerEndpoint = fmEndpointEnv
+	}
+
+	option := fleetmanager.OptionFromEnv()
+	auth, err := fleetmanager.NewStaticAuth(fleetmanager.StaticOption{StaticToken: option.Static.StaticToken})
+	Expect(err).ToNot(HaveOccurred())
+	client, err := fleetmanager.NewClient(fleetManagerEndpoint, auth)
+	Expect(err).ToNot(HaveOccurred())
+
+	Describe("should run ACS operators", func() {
+
+		It("should deploy single operator", func() {
+			ctx := context.Background()
+			oneOperatorVersionConfig := `
+- gitref: 4.1.1
+  image: quay.io/rhacs-eng/stackrox-operator:4.1.1`
+			err := updateOperatorConfig(ctx, oneOperatorVersionConfig)
+			Expect(err).To(BeNil())
+
+			Eventually(validateOperatorDeployment(ctx, "4.1.1", "quay.io/rhacs-eng/stackrox-operator:4.1.1")).
+				WithTimeout(waitTimeout).
+				WithPolling(defaultPolling).
+				Should(Succeed())
+
+		})
+
+		It("should deploy two operators", func() {
+			ctx := context.Background()
+			twoOperatorVersionConfig := `
+- gitref: 4.1.1
+  image: quay.io/rhacs-eng/stackrox-operator:4.1.1
+- gitref: 4.1.2
+  image: quay.io/rhacs-eng/stackrox-operator:4.1.2`
+			err := updateOperatorConfig(ctx, twoOperatorVersionConfig)
+			Expect(err).To(BeNil())
+
+			Eventually(validateOperatorDeployment(ctx, "4.1.1", "quay.io/rhacs-eng/stackrox-operator:4.1.1")).
+				WithTimeout(waitTimeout).
+				WithPolling(defaultPolling).
+				Should(Succeed())
+			Eventually(validateOperatorDeployment(ctx, "4.1.2", "quay.io/rhacs-eng/stackrox-operator:4.1.2")).
+				WithTimeout(waitTimeout).
+				WithPolling(defaultPolling).
+				Should(Succeed())
+			Eventually(getOperatorDeployments(ctx)).
+				WithTimeout(waitTimeout).
+				WithPolling(defaultPolling).
+				Should(HaveLen(2))
+
+		})
+
+		It("should run only the latest operator", func() {
+			ctx := context.Background()
+			twoOperatorVersionConfig := `
```

Comment:
imho it is sufficient because the context is clear and not multiple variables are used
```suggestion
			operatorConfig := `
```